module/VuFind/src/VuFind/Search/Base/Params.php

```diff
@@ -1064,23 +1064,24 @@ public function getFilterList($excludeCheckboxFilters = false)
     }
 
     /**
-     * Check for delimited facets -- if $field is a delimited facet field,
-     * process $displayText accordingly. Return the appropriate display value.
+     * Get a display text for a facet field.
      *
-     * @param string $field       The facet
-     * @param string $displayText The facet value
+     * @param string $field Facet field
+     * @param string $value Facet value
      *
      * @return string
      */
-    public function checkForDelimitedFacetDisplayText($field, $displayText)
+    public function getFacetValueDisplayText(string $field, string $value): string
```

Comment:
@EreMaijala, since the return value from this method may be subject to additional post-processing (for hierarchical and/or translated facets), should we rename this method to something like `getFacetValueRawDisplayText`?  Or should we pass the `$hierarchical` and `$translated` flags to this method and do additional processing in here to better encapsulate the logic?

Reply:
i like `getFacetValueRawDisplayText` better than encapsulating everything in it. The contexts where it's called are a bit different, so the additional processing is also different (e.g. whether to display all hierarchy levels).

Reply:
Okay, method renamed. I also noticed that formatting is only applied to facets, not to filters; this feels to me like an oversight rather than a decision, so I've created another support method for facet translation (see d0632fb) in order to make it easier to be consistent. This also allowed me to significantly compact some of the code. However, this is breaking a test, and I'm currently having trouble figuring out why... Investigating now.

Reply:
Broken test was related to a typehint prematurely converting a TranslatableString object into a string. Fixed in 71e9566, and thanks to @EreMaijala for pointing me in the right direction!